src/main/java/org/opentripplanner/routing/edgetype/TripPattern.java

```diff
@@ -69,6 +71,21 @@
     private static final int NO_PICKUP = 1;
     //private static final int FLAG_BIKES_ALLOWED = 32;
 
+    /**
+     * Currently used for NeTEx id
+     * TODO OTP2 - Merge with code or document why it should not be merged with code.
+     */
+    public FeedScopedId id;
+
+    /**
+     * The short unique identifier for this trip pattern,
+     * generally in the format Agency:RouteId:DirectionId:PatternNumber.
+     */
+    public String code;
```

Comment:
When importing GTFS data, this code should probably become the FeedScopedId of the pattern.